contracts/utils/ReentrancyGuard.sol

```diff
@@ -17,7 +17,10 @@ pragma solidity ^0.6.0;
  * https://blog.openzeppelin.com/reentrancy-after-istanbul/[Reentrancy After Istanbul].
  */
 contract ReentrancyGuard {
-    bool private _notEntered;
+    uint256 private constant _NOT_ENTERED = 1;
+    uint256 private constant _ENTERED = uint256(int256(-1));
```

Comment:
Is there a reason why you used this number in particular? It should be cheaper to use a smaller number because it can be pushed using `PUSH1` rather than `PUSH32`, which also results in smaller bytecode and is cheaper to deploy.

I think we can safely use the number 2 instead.

Reply:
I believe the reason why @BrendanChou used 0xff..ff is to make this backwards compatible (in a contract upgrade sense), by making sure the value of the reentrancy counter couldn't ever be equal to `_ENTERED`.

Reply:
For that to happen there would have to be a "reentrant" upgrade, right? And in any case any number higher than 1 would have the same "entered" meaning so I think the behavior would remain the same, no?

Reply:
The old ReentrancyGuard had a counter with the number of times it had executed. For `require(_status != _ENTERED)` to pass, you'd want to make sure `_status` (the counter) is always different from `_ENTERED`, hence 0xff..ff.

Reply:
Given that the plan isn't for this to be a storage-compatible upgrade from the v2.x code, I'd say we should go ahead and change these values to 1 and 2 so that they fit in a PUSH1. Can we use `enum`s for this, or will that incur added gas costs? 

I believe added range checks are performed on `enum` assignment, but I'd hope that's not the case when assigning constants.

Reply:
Range checks are often performed on enum assignments unfortunately. I'm not 100% sure about it doing it with constants

Reply:
As we're already not using `bool` or `enum` for storage, I see little point in making the constants into an enum. My 2 cents

Reply:
Ah true an `enum` would use `uint8` under the hood, silly me. Disregard that.